File: app/src/main/java/app/vodio/com/vodio/fragments/ActuFragment.java
package app.vodio.com.vodio.fragments;

import android.os.Bundle;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.Toast;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;

import app.vodio.com.vodio.R;
import app.vodio.com.vodio.beans.Vod;
import app.vodio.com.vodio.database.VodMapper;
import app.vodio.com.vodio.utils.MyAsyncTask;
import app.vodio.com.vodio.utils.OnCompleteAsyncTask;
import app.vodio.com.vodio.views.VodAdapter;


public class ActuFragment extends AbstractFragment {

    private ListView listPosts;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.fragment_actu, container, false);
        listPosts = v.findViewById(R.id.listOfPost);
        // Inflate the layout for this fragment
        return v;
    }

    @Override
    public void onResume() {
        super.onResume();
        final List<Vod> list = new ArrayList<>();
        final ArrayAdapter<Vod> adapter = new VodAdapter(getContext(), android.R.layout.simple_expandable_list_item_1, list);
        listPosts.setAdapter(adapter);
        MyAsyncTask<List<Vod>> taskActus = new VodMapper.GetListVodTask(new OnCompleteAsyncTask() {
            @Override
            public void onSuccess(Object obj) {
                // stop load view
                List<Vod> list = (List<Vod>)obj;
                adapter.addAll(list);
            }
            @Override
            public void onFail() {
                // stop load view
            }
        });
        // load view
        taskActus.execute();
    }

    @Override
    public void onClick(View view) {

    }

    private List<String> getList(){
        List<String> l = new ArrayList<>();
        for(int i = 0 ; i < 20 ; i++ ){
            l.add("Element "+i);
        }
        return l;
    }
}

File: app/src/main/java/app/vodio/com/vodio/fragments/ActuFragment.kt
package app.vodio.com.vodio.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.lifecycle.Observer
import androidx.recyclerview.widget.LinearLayoutManager
import app.vodio.com.vodio.R
import app.vodio.com.vodio.beans.Vod
import app.vodio.com.vodio.database.retrofit.repositories.LoginRepository
import app.vodio.com.vodio.fragments.utils.CollectionView
import app.vodio.com.vodio.models.VodsPresenter
import app.vodio.com.vodio.models.VodsPresenterFactory
import app.vodio.com.vodio.views.VodAdapter
import kotlinx.android.synthetic.main.fragment_actu.*
import java.util.*


class ActuFragment : AbstractFragment(), CollectionView<Vod> {

    private val vods = ArrayList<Vod>()
    private var adapter : VodAdapter? = null
    private var idVodProvided = ArrayList<Int>()

    private var vodPresenter : VodsPresenter? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_actu, container, false)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        initDataModels()
    }

    override fun onResume() {
        super.onResume()
        attachObservers()
        configureAdapter()
        configureOnRefresh()
    }

    override fun onPause() {
        super.onPause()
        detachObservers()
    }

    override fun showCollectionError(message: String) {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun showCollection(t : List<Vod>){
        loadingVods.visibility = View.GONE
        vods.clear()
        for (vod in t) {
            vods.add(vod)
            idVodProvided.add(vod.idVod)
            adapter!!.notifyItemInserted(vods.size )
        }
        refreshlayout.isRefreshing = false
    }

    override fun showCollectionNoData(){
        loadingVods.visibility = View.GONE
        noVodProvided.visibility = View.VISIBLE
        refreshlayout.isRefreshing = false
    }

    override fun showCollectionLoading(){
        loadingVods.visibility = View.VISIBLE
        noVodProvided.visibility = View.GONE
    }

    private fun attachObservers(){
        vodPresenter!!.loadFeedVods(LoginRepository.getInstance(context!!).getLoggedUser().value!!.login!!)
    }

    private fun detachObservers(){
        //vodModel!!.getFeedVods().removeObserver(vodObserver)
    }

    private fun configureOnRefresh(){
        refreshlayout.setOnRefreshListener {
            refreshlayout.isRefreshing = true
            vodPresenter!!.loadFeedVods(LoginRepository.getInstance(context!!).getLoggedUser().value!!.login!!)
        }
        refreshWhenEmptyActu.setOnClickListener {
            refreshlayout.isRefreshing = true
            vodPresenter!!.loadFeedVods(LoginRepository.getInstance(context!!).getLoggedUser().value!!.login!!)
        }
    }

    private fun configureAdapter(){
        if(adapter == null) {
            adapter = VodAdapter(vods, context!!, this)
        }
        listOfPost.adapter = adapter
        listOfPost.layoutManager = LinearLayoutManager(context)
    }



    private fun nonEmptyVodsUiMode(){
        loadingVods.visibility = View.GONE
        noVodProvided.visibility = View.GONE
        refreshlayout.visibility =  View.VISIBLE
    }

    private fun initDataModels(){
        vodPresenter = VodsPresenterFactory(context!!).create()
        vodPresenter!!.attachView(this, lifecycle)
    }

    override fun onClick(view: View) {}

    override fun getIconId(): Int {
        return iconMapping.idIcon
    }

    object iconMapping{
        const val idIcon = R.id.actu_navbar
    }

}
